src/components/EditSections/EditExtendedInfo/CreateResetPasswordControl/CreateResetPasswordControl.js

```diff
@@ -37,17 +40,73 @@ class CreateResetPasswordControl extends React.Component {
       fetch: false,
       throwErrors: false,
     },
-    isLocalPasswordSet: {
+    credentials: {
       type: 'okapi',
-      path: 'authn/credentials-existence?userId=!{userId}',
-      permissionsRequired: 'login.credentials-existence.get',
+      path: 'authn/credentials?query=(userId=!{userId})',
     },
   });
 
-  state = {
-    showModal: false,
-    link: '',
-  };
+  constructor(props) {
+    super(props);
+    this.state = {
+      showModal: false,
+      isLocalPasswordSet: false,
+      link: '',
+    };
+
+    this._isMounted = false;
+  }
+
+  componentDidMount() {
+    this._isMounted = true;
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this._isMounted && (this.props.resources.credentials !== prevProps.resources.credentials)) {
+      this.checkPasswordExistence();
+    }
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  checkPasswordExistence = async () => {
+    /*
+    We need to differ the user WITH setted password from user WITHOUT password.
+    When user was created by default setted password is '' (empty string).
+    When we edit user, we receive the object with credentials.
+    Then we encrypt the '' with received salt from credentials, and if hash is simular,
+    then we know that password was not set for the current user yet, then display 'Create password',
+    otherwise 'Reset password' link.
+    */
+    const {
+      resources: {
+        credentials,
+      },
+    } = this.props;
+    const credentialsRecord = credentials?.records[0]?.credentials[0] ?? {};
+
+    if (isEmpty(credentialsRecord)) {
+      return;
+    }
+
+    const normalizeSalt = Buffer.from(credentialsRecord.salt, 'hex');
+
+    await crypto.pbkdf2('', normalizeSalt, 1000, 32, 'sha1', (err, derivedKey) => {
```

Comment:
I suggest calling then pbkdf2Sync so there won't be the need for the await stuff.